Tella/Data/Networking/Repositories/UwaziServerRepositories.swift

```diff
@@ -78,6 +124,136 @@ struct UwaziServerRepository: WebRepository {
             .eraseToAnyPublisher()
     }
 }
+extension UwaziServerRepository {
+    /// Return a collection of CollectedTemplate which is used to created after all the manpulation is done to prepare the template data for storage
+    /// - Parameters:
+    ///   - server: Server Object to get the information about the Uwazi server
+    ///   - locale: UwaziLocale Object that has the locale information about language that the user selected when adding a new Uwazi server
+    /// - Returns: Collection of CollectedTemplate
+    func handleTemplate(server: Server, locale: UwaziLocale) async throws  -> AnyPublisher<[CollectedTemplate], Error> {
+        if let serverID = server.id, let serverURL = server.url {
+            let cookieList = [server.accessToken ?? "", locale.locale ?? ""]
+            let getTemplate = UwaziServerRepository().getTemplate(serverURL: serverURL, cookieList: cookieList)
+            let getSetting = UwaziServerRepository().getSettings(serverURL: serverURL, cookieList: cookieList)
+            let getDictionary = UwaziServerRepository().getDictionaries(serverURL: serverURL, cookieList: cookieList)
+            let getTranslation = UwaziServerRepository().getTranslations(serverURL: serverURL, cookieList: cookieList)
+
+            return Publishers.Zip4(
+                getTemplate,
+                getSetting,
+                getDictionary,
+                getTranslation
+            )
+            .tryMap({ (templateResult, settings, dictionary, translationResult) in
+                let templates = templateResult.rows
+                let translations = translationResult.rows
+                let dictionary = dictionary.rows
+                let settings: UwaziSetting = settings
+
+                // Maps the options to the property of the template
+                self.handleMapping(templates, dictionary)
+                // Check whether the server instance is public and if public then only use the whitelisted templates are added to resultTemplates
+                let resultTemplates = self.getAllowedTemplates(server: server, settings: settings, templates: templates)
+                self.translate(locale: locale, resultTemplates: resultTemplates, translations: translations)
+                let originalTemplate = resultTemplates.map { template in
+                    return CollectedTemplate(serverId: serverID,
```

Comment:
I think It's better to map the original template in the view model, (mabe later we will add an other layer that implements the call to the API and at the same time save to database)